idea/src/org/jetbrains/kotlin/idea/intentions/ConvertLambdaToReferenceIntention.kt

```diff
@@ -94,17 +102,23 @@ open class ConvertLambdaToReferenceIntention(textGetter: () -> String) : SelfTar
         }
 
         if (!descriptorHasReceiver && explicitReceiver != null && calleeDescriptor !is ClassConstructorDescriptor) return false
-        val noBoundReferences = !callableExpression.languageVersionSettings.supportsFeature(LanguageFeature.BoundCallableReferences)
+        val noBoundReferences = !languageVersionSettings.supportsFeature(LanguageFeature.BoundCallableReferences)
         if (noBoundReferences && descriptorHasReceiver && explicitReceiver == null) return false
 
         val callableArgumentsCount = (callableExpression as? KtCallExpression)?.valueArguments?.size ?: 0
-        if (calleeDescriptor.valueParameters.size != callableArgumentsCount) return false
-        val lambdaMustReturnUnit =
-            if (lambdaParameterType?.isFunctionType == true) lambdaParameterType.getReturnTypeFromFunctionType().isUnit() else false
-        if (lambdaMustReturnUnit) {
-            calleeDescriptor.returnType.let {
-                // If Unit required, no references to non-Unit callables
-                if (it == null || !it.isUnit()) return false
+        if (languageVersion >= LanguageVersion.KOTLIN_1_4) {
```

Comment:
This is also not a correct check -- this feature resides under the `LanguageFeature.FunctionReferenceWithDefaultValueAsOtherType` flag, which also can be disabled independently from the language version